### PR TITLE
Retrofit 2

### LINKED ISSUE #175: Switch to Retrofit 2

### PR DESCRIPTION
**Pull request by mattprecious**
Closes #175

### CODE REVIEW

File: src/internalDebug/java/com/jakewharton/u2020/data/api/MockGithubService.java [hunk 1]
```diff
@@ -53,6 +55,6 @@
       response = new RepositoriesResponse(items);
     }
 
-    return Observable.just(response);
+    return Observable.just(Result.response(Response.success(response)));
```
Comment (JakeWharton) [inline src/internalDebug/java/com/jakewharton/u2020/data/api/MockGithubService.java]: I think there's a convenience overload for `Result.response` which just lets you had it the `response` without wrapping.

File: src/internalDebug/java/com/jakewharton/u2020/data/prefs/LongPreference.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
+package com.jakewharton.u2020.data.prefs;
+
+import android.content.SharedPreferences;
+
+public class LongPreference {
```
Comment (JakeWharton) [inline src/internalDebug/java/com/jakewharton/u2020/data/prefs/LongPreference.java]: final

**Comment by JakeWharton** [comment]
looks good

File: src/internalDebug/java/com/jakewharton/u2020/data/api/MockGithubService.java (hunk 1 above)
Comment (mattprecious) [inline src/internalDebug/java/com/jakewharton/u2020/data/api/MockGithubService.java]: Nope. That's what you had before I took over the branch and it didn't compile.

Comment (JakeWharton) [inline src/internalDebug/java/com/jakewharton/u2020/data/api/MockGithubService.java]: Hmmm, I must've killed it then.

**Pull request merged**